staging/src/k8s.io/apiserver/pkg/authentication/authenticatorfactory/delegating.go

```diff
@@ -49,13 +50,21 @@ type DelegatingAuthenticatorConfig struct {
 	// ClientCAFile is the CA bundle file used to authenticate client certificates
 	ClientCAFile string
 
+	// SkippedPaths are HTTP paths which are excluded from authentication. They can be plain
+	// paths or end in * in which case prefix-match is applied. A leading / is optional.
+	SkippedPaths []string
+
 	RequestHeaderConfig *RequestHeaderConfig
 }
 
 func (c DelegatingAuthenticatorConfig) New() (authenticator.Request, *spec.SecurityDefinitions, error) {
 	authenticators := []authenticator.Request{}
 	securityDefinitions := spec.SecurityDefinitions{}
 
+	if len(c.SkippedPaths) > 0 {
+		authenticators = append(authenticators, path.NewAuthenticator(c.SkippedPaths))
```

Comment:
I didn't expect this to be required for authentication... if you allow anonymous requests, a request with no client cert or token will already be authenticated as the anonymous user (for all paths, not just /healthz)

Reply:
Even if authenticated, we want to exclude certain paths in case the kube-apiserver is down. Otherwise, with a client cert you can query `/healthz` in that case.

Maybe for authn this would be fine. Authz still needs the skipped paths mechanism.